server/src/main/java/org/elasticsearch/index/mapper/MapperBuilderContext.java

```diff
@@ -64,4 +70,12 @@ public boolean isSourceSynthetic() {
     public boolean isDataStream() {
         return isDataStream;
     }
+
+    /**
+     * Are these field mappings being built dimensions?
+     */
+    public boolean parentFieldContainsDimensions() {
+        return parentFieldContainsDimensions;
```

Comment:
Sorry for another naming comment :) 
What about `parentFieldEnablesDimension`?  If `time_series_dimension` is configured on a passthrough about then this automatically makes any number, keyword, ip field a dimension too. So a parent passthrough field essentially enabled the whether a field is a time series dimension.

Reply:
Changed to `parentObjectContainsDimension`, matching the `containsDimension` param of `PassThroughObject`. `parentFieldEnablesDimension` is not accurate imho, this doesn't just enable but enforce defining subfields as dimensions.